github/resource_github_repository_collaborators_test.go

```diff
@@ -52,9 +42,61 @@ func TestAccGithubRepositoryCollaborators(t *testing.T) {
 					permission = "push"
 				}
 			}
-		`, repoName, inOrgUser)
+		`, repoName, testAccConf.testExternalUser)
 
-		orgConfig := fmt.Sprintf(`
+		resource.Test(t, resource.TestCase{
+			PreCheck:          func() { skipUnauthenticated(t) },
+			ProviderFactories: providerFactories,
+			Steps: []resource.TestStep{
+				{
+					Config: config,
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttrSet("github_repository_collaborators.test_repo_collaborators", "user.#"),
+						resource.TestCheckResourceAttrSet("github_repository_collaborators.test_repo_collaborators", "team.#"),
+						resource.TestCheckResourceAttr("github_repository_collaborators.test_repo_collaborators", "user.#", "1"),
+						resource.TestCheckResourceAttr("github_repository_collaborators.test_repo_collaborators", "team.#", "0"),
+						func(state *terraform.State) error {
+							owner := testAccConf.owner
+
+							collaborators := state.RootModule().Resources["github_repository_collaborators.test_repo_collaborators"].Primary
+							for name, val := range collaborators.Attributes {
+								if strings.HasPrefix(name, "user.") && strings.HasSuffix(name, ".username") && val != testAccConf.testExternalUser {
+									return fmt.Errorf("expected user.*.username to be set to %s, was %s", testAccConf.testExternalUser, val)
+								}
+								if strings.HasPrefix(name, "user.") && strings.HasSuffix(name, ".permission") && val != "push" {
+									return fmt.Errorf("expected user.*.permission to be set to push, was %s", val)
+								}
+							}
+
+							invites, _, err := conn.Repositories.ListInvitations(context.TODO(), owner, repoName, nil)
```

Comment:
Do we have access to `t` in here?
```suggestion
							invites, _, err := conn.Repositories.ListInvitations(t.Context(), owner, repoName, nil)
```